src/enumerate.rs

```diff
@@ -14,8 +14,13 @@ use libparsec::{
 
 use crate::protocol::{ProtocolErrorFields, ProtocolResult};
 
+// #[non_exhaustive] macro must be set for every enum,
```

Comment:
```suggestion
// #[non_exhaustive] macro must be set for every enum like type,
```

Because here `non_exhaustive` is set on `rust` `struct` and this could be misleading as we're using them like an enum in the python context.